stream.go

```diff
@@ -167,6 +164,12 @@ func (s *StreamEntry) UnmarshalRESP(br *bufio.Reader) error {
 		}
 	}
 
+	if ah.N == -1 {
```

Comment:
I believe this if check should still come before the initialization of `Fields` above, since that's using `ah.N` to guess the map size in one of the cases. If `ah.N` is negative that would... panic? It wouldn't do anything good. Is there a reason you moved this check here?

Reply:
Oh yeah, that'd be a problem. I moved the check down here, because I wanted to make use of this loop that clears the Fields map:
```
for k := range s.Fields {
	delete(s.Fields, k)
}
```
If `ah.N == -1` we should probably still clear the map.

Reply:
Changed it. How's that?

Reply:
We could also go with
```
	if s.Fields != nil {
		for k := range s.Fields {
			delete(s.Fields, k)
		}
	}

	if ah.N == -1 {
		return nil
	} else if ah.N%2 != 0 {
		return errInvalidStreamEntry
	}

	if s.Fields == nil {
		s.Fields = make(map[string]string, ah.N/2)
	}
```

Reply:
tbh I like this way ^ better. It's less duplicate code, and easier to read. I think the `if s.Fields != nil` isn't even necessary, ranging over a nil map is like ranging over a nil slice.

Reply:
cool, changed